### DISCUSSION TITLE
SNAP Compatibility

### DISCUSSION THREAD

**Discussion by Reefs-Endeavour** [HandyHat/ha-hildebrandglow-dcc#308] [General]
I've recently upgrade all the components in my Home Assistant setup and the glow integration is no longer working.

Looking at the logs I have the following error:

    from glowmarkt import BrightClient
ModuleNotFoundError: No module named 'glowmarkt'

I've tried installing Bright Client, but I think I have other issues as I'm running these as snaps

Name                 Version       Rev    Tracking       Publisher       Notes
home-assistant-snap  2023.1.7      547    latest/stable  giaever-online  -
mosquitto            2.0.15        776    latest/stable  mosquitto✓      -
snapd                2.58          17950  latest/stable  canonical✓      snapd

Do I require an updated version of home-assistant-snap with BrightClient support or is there something else I'm missing here?

Thanks

**Comment by HandyHat** [comment]
Hey @Reefs-Endeavour, this is an odd one. Home Assistant should automatically install the library when you add the custom component. What version of this integration are you running?

**Comment by Reefs-Endeavour** [comment]
Home Assistant 2023.2.2
HACS 1.30.1
Hildebrand Glow (DCC) v1.0.3

**Comment by HandyHat** [comment]
Interesting...
If you have cli access to home assistant you can run `pip install glowmarkt` to revolve this, but as it's a snap I suspect you won't

Can you try removing the integration completely (i.e., uninstalling it via HACS, not just removing the config entry), restarting, and then reinstalling it via HACS? Your sensors will all keep their history.

**Comment by Reefs-Endeavour** [comment]
Thanks,

I get the following:-

ERROR: Could not find a version that satisfies the requirement glowmarkt (from versions: none)
ERROR: No matching distribution found for glowmarkt

Also, I've removed and re-added the integration, I've also cleared the follwoing location of files:-

/var/snap/home-assistant-snap/551/custom_components/hildebrandglow_dcc/

Adding the integration to HACS brings them back as expected, but I still can't fix the glowmarkt issue

And for further info:-

pip 23.0 from /usr/local/lib/python3.10/dist-packages/pip (python 3.10)

**Comment by HandyHat** [comment]
Apologies - the command is `pip install pyglowmarkt`

The issue is that you will have to do this every time the version of the library changes, which isn't a great experience.

**Reply by HandyHat** [reply]
Ah, it looks like this is indeed an issue with the snap. See: https://git.giaever.org/joachimmg/home-assistant-snap/issues/10 and https://git.giaever.org/joachimmg/home-assistant-snap/issues/11

Unfortunately I think there's very little I can do to resolve this

**Reply by Reefs-Endeavour** [reply]
Thanks for looking in to it and confirming.  I think I may just rebuild without snaps

**Reply by Reefs-Endeavour** [reply]
I’ve rebuilt my setup. It’s now running Debian Supervisor and the Hildebrand integration setup without any issue.

I’ve logged on and I’m pulling back data. I can see rate and standing charge information correctly. Electric appears to be pulling back the wrong daily amount and daily costs. Gas isn’t pulling any usage, but is pulling back an incorrect daily cost.
